### PR TITLE
PECG Report Indicator

### PR DESCRIPTION
**Pull request by IipperMangesh**
Added PECG Report Indicator "Number of new PLWHA placed on ARV who took treatment in the month".

### CODE REVIEW

**Comment by tmvumbi2** [comment]
@IipperMangesh  the feature branch needs to be directly created on the Jembi's repository (here: https://github.com/jembi/cameroon-bahmni-config): 
- the feature branch's name should use the ticket number (I.e. COMRS-124);
- a PR needs to be added to merge the feature branch into the dev branch;
- each commit needs to have the ticket's number on the subject, a short subject (i.e max 70 chars) a description explaining what's done and the motivation

**Comment by tmvumbi2** [comment]
@IipperMangesh On the generated excel report, only the sheet `Report` was populated (the first sheet `Sheet` had zeroes everywhere). Note that I am testing on Ubuntu with LibreOffice Calc; don't know if this might be an issue only there

**Comment by tmvumbi2** [comment]
@IipperMangesh I would suggest we use a lint tool to format the SQL so that it is more readable; as an example, below is the lint result from http://sqllint.com/: 


image

File: openmrs/apps/reports/PECG_REPORT/sql/pecg.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+         CASE WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) >= 50 AND p.gender = 'M'
+         THEN COUNT(1)  END AS 'GrtThan50YrsMale',
+         CASE WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) >= 50 AND p.gender = 'F'
+         THEN COUNT(1)  END AS 'GrtThan50YrsFemale'
+    FROM (
+
+           select distinct p.person_id,pp.date_enrolled  from
+   visit v join person_name pn on v.patient_id = pn.person_id and pn.voided = 0 and v.voided=0
+   join patient_identifier pi on v.patient_id = pi.patient_id and pi.voided=0
+   join patient_identifier_type pit on pi.identifier_type = pit.patient_identifier_type_id
```
[the remark this answers is not in the document]
Comment (IipperMangesh) [reply openmrs/apps/reports/PECG_REPORT/sql/pecg.sql]: I have removed the patient_identifier_type table join from the query.

File: openmrs/apps/reports/PECG_REPORT/sql/pecg.sql [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+   join person p on p.person_id = v.patient_id  and p.voided=0
+   join patient_program pp on pp.patient_id=p.person_id and pp.voided=0 and pp.date_completed is null
+   and cast(pp.date_enrolled AS DATE) between '#startDate#'AND '#endDate#'
+   join program pr on pr.program_id=pp.program_id and pr.name like '%HIV Program%'
+
+
+            ) AS numberOfPLARTinCareNewlyEnrolled
```
[the remark this answers is not in the document]
Comment (IipperMangesh) [reply openmrs/apps/reports/PECG_REPORT/sql/pecg.sql]: Sure , I have updated the aliases as Listofpatient…

File: openmrs/apps/reports/PECG_REPORT/sql/pecg.sql [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+               THEN '10-14 Yrs M'
+               WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) BETWEEN 10 AND 14 AND p.gender = 'F'
+               THEN '10-14 Yrs F'
+               WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) BETWEEN 15 AND 19 AND p.gender = 'M'
+               THEN '15-19 Yrs M'
+               WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) BETWEEN 15 AND 19 AND p.gender = 'F'
+               THEN '15-19 Yrs F'
+               WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) BETWEEN 20 AND 24 AND p.gender = 'M'
+               THEN '20-24 Yrs M'
+               WHEN timestampdiff(YEAR,p.birthdate,numberOfPLARTinCareNewlyEnrolled.date_enrolled) BETWEEN 20 AND 24 AND p.gender = 'M'
```
[the remark this answers is not in the document]
Comment (IipperMangesh) [reply openmrs/apps/reports/PECG_REPORT/sql/pecg.sql]: ok

File: openmrs/apps/reports/reports.json [hunk 4]
```diff
@@ -26,6 +26,14 @@
             ]
         }
     },
+    "NumberARTinitiationThisMonth": {
+        "name": "PECG Report",
+        "type": "MRSGeneric",
```
[the remark this answers is not in the document]
Comment (IipperMangesh) [reply openmrs/apps/reports/reports.json]: The report type is correct as "type": "MRSGeneric", the "MRSGeneric" is provided by Bahmni. 
For reference you have look here:
https://bahmni.atlassian.net/wiki/spaces/BAH/pages/18415871/Configure+Reports#ConfigureReports-3.CustomSQL

**Pull request closed without merging**